extensions/mssql/src/sqlToolsServer.ts

```diff
@@ -143,30 +147,34 @@ function generateHandleServerProviderEvent() {
 }
 
 function getClientOptions(context: AppContext): ClientOptions {
+	const features = [
+		// we only want to add new features
+		...SqlOpsDataClient.defaultFeatures,
+		TelemetryFeature,
+		AccountFeature,
+		AgentServicesFeature,
+		SerializationFeature,
+		SqlAssessmentServicesFeature,
+		SchemaCompareService.asFeature(context),
+		LanguageExtensionService.asFeature(context),
+		DacFxService.asFeature(context),
+		CmsService.asFeature(context),
+		SqlAssessmentService.asFeature(context),
+		NotebookConvertService.asFeature(context),
+		ProfilerFeature,
+		SqlMigrationService.asFeature(context)
+	];
+	if (nativeCredentialsEnabled()) {
```

Comment:
Add native credential service as a feature if the setting is set, otherwise use the server